plugins/foreman_remote_execution/nightly/index.md

```diff
@@ -646,6 +646,32 @@ removes it on target host.
 In this case, previewing of the template would immediately trigger destroy
 method and therefore it's wrapped in preview condition.
 
+## 4.5 Plugin developer API
+
+Plugins can leverage Remote Execution to perform a particular job on a host by
+defining a Remote Execution Feature and the inputs required.
+
+The feature can be registered in the plugin initializer:
+
+```
+RemoteExecutionFeature.register(:reprovision, N_("Reprovision"),
+                                :description => "Reprovision the host via script",
+                                :provided_inputs => ["script"])
+```
+
+The mapping between the feature and which templates to use can be made in the
+UI via 'Administer -> Remote Execution Features'.  Or, when seeding a template,
+there is a new metadata item `feature`, which allows to assign some template as
+default to the feature.
+
+When the plugin wants to execute the paritcular feature they can use:
+
+```
+composer = JobInvocationComposer.for_feature(:reprovision, host, :script => "grubby...")
```

Comment:
It can be simplified with `JobInvocationComposer.for_feature(:reprovision, host, :script => "grubby...").trigger!`.